examples/web_integration_service/library_api/c++11/web_integration_service_library_api.cxx

```diff
@@ -0,0 +1,95 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <csignal>
+#include <map>
+
+#include "ndds/ndds_cpp.h"
+#include "rti/webdds/Service.hpp"
+
+static bool shutdown_requested;
+
+void stop_handler(int value)
+{
+    shutdown_requested = true;
+    std::cout << "preparing to shut down...\n";
+}
+void setup_signal_handlers()
+{
+    shutdown_requested = false;
+    signal(SIGINT, stop_handler);
+    signal(SIGTERM, stop_handler);
+}
+
+
+void service_main(std::string cfg_name, std::string cfg_file)
+{
+    rti::webdds::ServiceProperty service_property;
+
+    service_property.cfgfile(cfg_file).cfgname(cfg_name).document_root(
+            "../../../../web_integration_service/");
```

Comment:
Let's find a way to get the path to the source files (or binary dir) so we don't rely on the directory where we're running from.